subt/simulation.py

```diff
@@ -0,0 +1,159 @@
+import collections
+import datetime
+import inspect
+import math
+
+from threading import Thread, RLock
+
+from osgar.lib import quaternion
+
+try:
+    from colored import stylize, fg
+except ImportError:
+    def stylize(what, style):
+        return what
+    def fg(color_name):
+        return ""
+
+_saved_print = print
+
+def verbose(ok):
+    print(ok)
+    if ok:
+        globals()['print'] = _saved_print
+    else:
+        globals()['print'] = lambda *args: None
+
+
+def log(*args):
+    up = inspect.currentframe().f_back
+    caller = up.f_code.co_name
+    caller_class = up.f_locals['self'].__class__
+    color_name = getattr(caller_class, caller).color
+    print(stylize(f"{caller_class.__name__}.{caller}", fg(color_name)), *args)
+
+
+def clip(min_value, value, max_value):
```

Comment:
Not sure how you feel about external dependencies at this point. This is np.clip() and you may not need to define it yourself.